cannot convert ‘__vector(4) __bool int’ to ‘unsigned int’

From a [PPA build of the development branch in Ubuntu 20.04 on ppc64el:](https://launchpadlibrarian.net/520401223/buildlog_ubuntu-focal-ppc64el.macaulay2_1.17.2.1+git202101300204-0ppa202101300214~ubuntu20.04.1_BUILDING.txt.gz)

```shell
g++ -pipe -Wall -Wno-shadow -Wcast-qual -Wno-sign-conversion -Wno-sign-compare -Wno-parentheses -Wno-sign-compare  -fno-stack-protector -Wuninitialized -I/usr/include  -Wno-error=pragmas -std=gnu++14 -g -fdebug-prefix-map=/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204=. -fstack-protector-strong -Wformat -Werror=format-security -g3 -O2  -Wno-unused-local-typedefs -Wno-unknown-pragmas -std=c++17 -Wno-mismatched-tags -I. -DNDEBUG -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/include -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/include -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/usr-host/include  -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/include -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/include -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/usr-host/include  -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/include -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/include -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/usr-host/include  -Wdate-time -D_FORTIFY_SOURCE=2 -isystem /usr/include/libxml2   -DSING_NDEBUG -DOM_NDEBUG -DSING_NDEBUG -DOM_NDEBUG -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular   -I/usr/src/gtest  -Wdate-time -D_FORTIFY_SOURCE=2 -DNDEBUG  -I/usr/include/cdd -I/usr/include/eigen3 -I/usr/include -DBOOST_STACKTRACE_LINK -isystem /usr/include/libxml2   -DSING_NDEBUG -DOM_NDEBUG -DSING_NDEBUG -DOM_NDEBUG -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular   -I/usr/src/gtest  -Wdate-time -D_FORTIFY_SOURCE=2 -DNDEBUG  -I/usr/include/cdd -I/usr/include/eigen3 -I/usr/include -DBOOST_STACKTRACE_LINK -isystem /usr/include/libxml2   -DSING_NDEBUG -DOM_NDEBUG -DSING_NDEBUG -DOM_NDEBUG -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular   -I/usr/src/gtest  -Wdate-time -D_FORTIFY_SOURCE=2 -DNDEBUG  -I/usr/include/cdd -I/usr/include/eigen3 -I/usr/include -DBOOST_STACKTRACE_LINK -I../d -I./../c -Wno-fatal-errors  -c -Wno-cast-qual VectorArithmetic.cpp -o VectorArithmetic.o
In file included from MemoryBlock.hpp:8,
                 from VectorArithmetic.hpp:6,
                 from VectorArithmetic.cpp:1:
/usr/include/tbb/tbb.h:21:154: note: #pragma message: TBB Warning: tbb.h contains deprecated functionality. For details, please see Deprecated Features appendix in the TBB reference manual.
   21 | #pragma message("TBB Warning: tbb.h contains deprecated functionality. For details, please see Deprecated Features appendix in the TBB reference manual.")
      |                                                                                                                                                          ^
In file included from /usr/include/fflas-ffpack/fflas/fflas.h:69,
                 from /usr/include/fflas-ffpack/ffpack/ffpack.h:45,
                 from aring-zzp-ffpack.hpp:21,
                 from aring-translate.hpp:18,
                 from aring-glue.hpp:7,
                 from VectorArithmetic.hpp:10,
                 from VectorArithmetic.cpp:1:
/usr/include/fflas-ffpack/utils/fflas_memory.h: In function ‘__vector(4) __bool int FFLAS::alignable()’:
/usr/include/fflas-ffpack/utils/fflas_memory.h:48:16: error: cannot convert ‘bool’ to ‘__vector(4) __bool int’ in return
   48 |         return true ;
      |                ^~~~
/usr/include/fflas-ffpack/utils/fflas_memory.h: At global scope:
/usr/include/fflas-ffpack/utils/fflas_memory.h:53:17: error: template-id ‘alignable<Givaro::Integer*>’ for ‘unsigned int FFLAS::alignable()’ does not match any template declaration
   53 |     inline bool alignable<Givaro::Integer*>() {
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/fflas-ffpack/utils/fflas_memory.h:47:17: note: candidate is: ‘template<class Element> __vector(4) __bool int FFLAS::alignable()’
   47 |     inline bool alignable() {
      |                 ^~~~~~~~~
...
```

This is followed by many, many more warning and error messages.  See the log for more.

Same error message, but in a gmp header instead of a fflas-ffpack one, in [Ubuntu 18.04 (also ppc64el)](https://launchpadlibrarian.net/520401279/buildlog_ubuntu-bionic-ppc64el.macaulay2_1.17.2.1+git202101300204-0ppa202101300214~ubuntu18.04.1_BUILDING.txt.gz):

```shell
g++ -pipe -Wall -Wno-shadow -Wcast-qual -Wno-sign-conversion -Wno-sign-compare -Wno-parentheses -Wno-sign-compare  -fno-stack-protector -Wuninitialized -I/usr/include  -Wno-error=pragmas -std=gnu++14 -g -fdebug-prefix-map=/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204=. -fstack-protector-strong -Wformat -Werror=format-security -g3 -O2  -Wno-unused-local-typedefs -Wno-unknown-pragmas -std=c++17 -Wno-mismatched-tags -I. -DNDEBUG -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/include -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/include -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/usr-host/include  -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/include -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/include -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/usr-host/include  -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/include -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/include -I/<<BUILDDIR>>/macaulay2-1.17.2.1+git202101300204/M2/usr-host/include  -Wdate-time -D_FORTIFY_SOURCE=2 -isystem /usr/include/libxml2   -DSING_NDEBUG -DOM_NDEBUG -DSING_NDEBUG -DOM_NDEBUG -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular   -I/usr/src/gtest  -Wdate-time -D_FORTIFY_SOURCE=2 -DNDEBUG  -I/usr/include/cdd -I/usr/include/eigen3 -I/usr/include -DBOOST_STACKTRACE_LINK -isystem /usr/include/libxml2   -DSING_NDEBUG -DOM_NDEBUG -DSING_NDEBUG -DOM_NDEBUG -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular   -I/usr/src/gtest  -Wdate-time -D_FORTIFY_SOURCE=2 -DNDEBUG  -I/usr/include/cdd -I/usr/include/eigen3 -I/usr/include -DBOOST_STACKTRACE_LINK -isystem /usr/include/libxml2   -DSING_NDEBUG -DOM_NDEBUG -DSING_NDEBUG -DOM_NDEBUG -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular -I/usr/include/powerpc64le-linux-gnu/singular -I/usr/include/singular   -I/usr/src/gtest  -Wdate-time -D_FORTIFY_SOURCE=2 -DNDEBUG  -I/usr/include/cdd -I/usr/include/eigen3 -I/usr/include -DBOOST_STACKTRACE_LINK -I../d -I./../c -Wno-fatal-errors  -c -Wno-cast-qual VectorArithmetic.cpp -o VectorArithmetic.o
In file included from MemoryBlock.hpp:8:0,
                 from VectorArithmetic.hpp:6,
                 from VectorArithmetic.cpp:1:
/usr/include/tbb/tbb.h:21:154: note: #pragma message: TBB Warning: tbb.h contains deprecated functionality. For details, please see Deprecated Features appendix in the TBB reference manual.
 #pragma message("TBB Warning: tbb.h contains deprecated functionality. For details, please see Deprecated Features appendix in the TBB reference manual.")
                                                                                                                                                          ^
In file included from /usr/include/gmp++/gmp++.h:33:0,
                 from /usr/include/givaro/givinteger.h:19,
                 from /usr/include/fflas-ffpack/utils/fflas_memory.h:45,
                 from /usr/include/fflas-ffpack/fflas/fflas.h:71,
                 from /usr/include/fflas-ffpack/ffpack/ffpack.h:46,
                 from aring-zzp-ffpack.hpp:21,
                 from aring-translate.hpp:18,
                 from aring-glue.hpp:7,
                 from VectorArithmetic.hpp:10,
                 from VectorArithmetic.cpp:1:
/usr/include/gmpxx.h: In static member function ‘static __vector(4) __bool int __gmp_binary_equal::eval(mpz_srcptr, mpz_srcptr)’:
/usr/include/gmpxx.h:930:74: error: cannot convert ‘bool’ to ‘__vector(4) __bool int’ in return
   static bool eval(mpz_srcptr z, mpz_srcptr w) { return mpz_cmp(z, w) == 0; }
                                                                          ^
/usr/include/gmpxx.h: In static member function ‘static __vector(4) __bool int __gmp_binary_equal::eval(mpz_srcptr, long unsigned int)’:
/usr/include/gmpxx.h:933:32: error: cannot convert ‘bool’ to ‘__vector(4) __bool int’ in return
   { return mpz_cmp_ui(z, l) == 0; }
                                ^
/usr/include/gmpxx.h: In static member function ‘static __vector(4) __bool int __gmp_binary_equal::eval(mpz_srcptr, long int)’:
/usr/include/gmpxx.h:937:32: error: cannot convert ‘bool’ to ‘__vector(4) __bool int’ in return
   { return mpz_cmp_si(z, l) == 0; }
                                ^
/usr/include/gmpxx.h: In static member function ‘static __vector(4) __bool int __gmp_binary_equal::eval(mpz_srcptr, double)’:
/usr/include/gmpxx.h:941:31: error: cannot convert ‘bool’ to ‘__vector(4) __bool int’ in return
   { return mpz_cmp_d(z, d) == 0; }
                               ^
/usr/include/gmpxx.h: In static member function ‘static __vector(4) __bool int __gmp_binary_equal::eval(mpq_srcptr, mpq_srcptr)’:
/usr/include/gmpxx.h:946:31: error: cannot convert ‘bool’ to ‘__vector(4) __bool int’ in return
   { return mpq_equal(q, r) != 0; }
                               ^
/usr/include/gmpxx.h: In static member function ‘static __vector(4) __bool int __gmp_binary_equal::eval(mpq_srcptr, long unsigned int)’:
/usr/include/gmpxx.h:949:35: error: cannot convert ‘bool’ to ‘__vector(4) __bool int’ in return
   { return mpq_cmp_ui(q, l, 1) == 0; }
                                   ^
/usr/include/gmpxx.h: In static member function ‘static __vector(4) __bool int __gmp_binary_equal::eval(mpq_srcptr, long int)’:
/usr/include/gmpxx.h:953:35: error: cannot convert ‘bool’ to ‘__vector(4) __bool int’ in return
   { return mpq_cmp_si(q, l, 1) == 0; }
                                   ^
/usr/include/gmpxx.h: In static member function ‘static __vector(4) __bool int __gmp_binary_equal::eval(mpq_srcptr, double)’:
/usr/include/gmpxx.h:957:55: error: cannot convert ‘bool’ to ‘__vector(4) __bool int’ in return
   {  __GMPXX_TMPQ_D;    return mpq_equal (q, temp) != 0; }
                                                       ^
/usr/include/gmpxx.h: In static member function ‘static __vector(4) __bool int __gmp_binary_equal::eval(mpq_srcptr, mpz_srcptr)’:
/usr/include/gmpxx.h:961:31: error: cannot convert ‘bool’ to ‘__vector(4) __bool int’ in return
   { return mpq_cmp_z(q, z) == 0; }
                               ^
/usr/include/gmpxx.h: In static member function ‘static __vector(4) __bool int __gmp_binary_equal::eval(mpf_srcptr, mpf_srcptr)’:
/usr/include/gmpxx.h:965:74: error: cannot convert ‘bool’ to ‘__vector(4) __bool int’ in return
   static bool eval(mpf_srcptr f, mpf_srcptr g) { return mpf_cmp(f, g) == 0; }
                                                                          ^
/usr/include/gmpxx.h: In static member function ‘static __vector(4) __bool int __gmp_binary_equal::eval(mpf_srcptr, long unsigned int)’:
/usr/include/gmpxx.h:968:32: error: cannot convert ‘bool’ to ‘__vector(4) __bool int’ in return
   { return mpf_cmp_ui(f, l) == 0; }
...
```

--- a/M2/Macaulay2/d/Makefile.in
+++ b/M2/Macaulay2/d/Makefile.in
@@ -181,7 +181,7 @@ CFLAGS   += -Wunused-function
 CXXFLAGS += -Wunused-function
 
 CFLAGS   += -Wfatal-errors
-CXXFLAGS += -Wfatal-errors -std=c++17
+CXXFLAGS += -Wfatal-errors
 
 version.o : CXXFLAGS += -Wno-unused-local-typedefs
 

--- a/M2/Macaulay2/e/Makefile.in
+++ b/M2/Macaulay2/e/Makefile.in
@@ -9,7 +9,7 @@ PRE.cc   = $(COMPILE.cc) -E
 include @srcdir@/Makefile.files
 include Makefile.common
 CPPFLAGS := -I@srcdir@ $(CPPFLAGS) -I../d -I@srcdir@/../c -Wno-fatal-errors
-CXXFLAGS += -Wno-unknown-pragmas -std=c++17
+CXXFLAGS += -Wno-unknown-pragmas -std=gnu++17
 
 DOXYGEN_CONF_DIR :=$(SRCDIR)/doxygen-settings
 

